lib/ansible/modules/system/macos_pref.py

```diff
@@ -0,0 +1,372 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2018 - Matthias Hollerbach <mail@matthias-hollerbach.de>
+# Copyright (c) 2016 - John Calixto <john.calixto@nordstrom.com> and Nordstrom, Inc.
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: macos_pref
+author:
+    - Matthias Hollerbach (@kinglouie)
+    - John Calixto (@nordjc)
+short_description: Manipulates macOS preferences including complex data types.
+version_added: "2.6"
+description:
+    - This module allows users to create, read, update, and delete system and application preferences on macOS
+      including deeply nested dictionary values like those in the C(com.apple.finder) domain.
+    - It also provides a convenient C(merge) strategy for assignments in nested dictionaries so that users can very
+      specifically target nested keys without having to assign unchanged peer or parent values.
+requirements:
+    - Target machine should be running macOS
+    - PyObjC (preinstalled by Apple with the operating system)
+options:
+    domain:
+        description:
+            - The preference domain. E.g. C(com.apple.finder), C(/path/to/some.plist), C(NSGlobalDomain).
+        required: false
+        default: NSGlobalDomain
+    host:
+        description:
+            - The host on which the preference should apply.
+        required: false
+        default: anyHost
+        choices: ["anyHost", "currentHost"]
+    key:
+        description:
+            - The preference key.
+        required: true
+    value:
+        description:
+            - The value that will be set for the specified key.
```

Comment:
Document that if value is not provided, the preference value will be read instead. Might be useful to mention this in the main description too.

Reply:
Will update that 👍